docs/settings.md

```diff
@@ -0,0 +1,50 @@
+## `src/Settings.js`
+
+Autofill needs access to certain pieces of state before it can perform any page scanning, 
+therefor a 'Settings' object has been introduced to allow this state to be queried and updated before
+any page scanning occurs.
+
+## Properties
+
+`settings.featureToggles` 
+ 
+The platform in question will deliver feature toggles (boolean flags) based on device support, user preferences, or a combination of both. Inside Autofill we don't make any distinction about 'how' the toggles are set, we just ask the platform for them and then use them verbatim. This prevents additional logic from bleeding into the Autofill codebase.
+
+- `inputType_credentials` - whether the device can autofill credentials
+- `inputType_identities` - whether the device can autofill identities
+- `inputType_creditCards` - whether the device can autofill creditCards
+- `emailProtection` - whether the device supports email protection
+- `password_generation` - if the device can offer generated passwords
+- `credentials_saving` - if the device should offer to capture submitted form data to save
+
+---
+
+`settings.availableInputTypes` 
+
+Another set of boolean flags, this time indicating which data types the 
+current user can autofill. This is domain specific.
+
+- `credentials` - true if the user has credentials stored for the current domain
+- `identities` - true if the user has identities stored for the current domain
+- `creditCards` - true if the user has creditCards stored for the current domain
+- `email` - true if the user has logged into email protection - this one is *not* domain specific.
+  - note: `availableInputTypes.email` is only current used on android. In the future all platforms will migrate to this.
+
+## Methods
+
+`settings.refresh(availableInputTypesOverrides)`
```

Comment:
@GioSensation The idea with 'settings.refresh()' is that it may or may not include API calls, lookups from string replacements, defaults, or anything else. 

It's designed to be something that's safe to call whenever, to update 'state' of the user/application.

For example, when a user has signed into email protection it should be safe to call 'refresh()' and have it determine the next state.

Let me know if this answers your question.